projects/adc_periodic_reader/test/test_adc_periodc_reader.c

```diff
@@ -0,0 +1,69 @@
+#include "adc.h"
+#include "adc_periodic_reader.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "misc.h"
+#include "ms_test_helpers.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "test_helpers.h"
+#include "wait.h"
+
+static int count = 0;
+
+void callback_test_1(uint16_t data, PeriodicReaderId id, void *context);

```

Comment:
I actually had to forward declare them here since reader_settings1 is dependent on the callback_test1, but at the same time callback_test1 is also dependent on reader_settings1 due to the test I put in callback_test1